internal/repository/osbuild/osbuild.go

```diff
@@ -0,0 +1,77 @@
+package osbuild
+
+import (
+	"context"
+	"github.com/project-flotta/osbuild-operator/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -package=osbuild -destination=mock_osbuild.go . Repository
+type Repository interface {
```

Comment:
Are all of the methods needed? I would keep it lean and add new ones when needed.